### ISSUE TITLE
Questions about the loss function

### ISSUE THREAD

**Issue by tang12138-yyhh** [eml-eda/mixprec-pruning#5]
Hello, @bmots 
In my opinion, the loss function mentioned in your paper, such as equation (2), selects the appropriate quantization bit width for each layer of the network by considering the weight set W and the bit width selection parameter θ and the regularization term R.
![image](https://github.com/user-attachments/assets/e6c4825a-781b-4b91-808d-71c2effc4710)
A discrete probability distribution is obtained from a continuous bit width selection parameter, which is then used to compute the effective activation and weight tensors, through three sampling methods such as equation (3).
![image](https://github.com/user-attachments/assets/6132066e-db82-4991-93d9-6d6000dd6176)
Now, I have a question, I can not find the design of the loss function in the source code you provided, can you help me?

**Comment by bmots** [comment]
Hello @tang12138-yyhh ,

you can find the computation of the loss [here](https://github.com/eml-eda/mixprec-pruning/blob/main/icl_mixprec_training_cost_size.py#L54), where you have the sum of a task loss term and a regularization loss term, weighted by the regularization strength hyperparameter.
By default, the regularization term corresponds to the size of the model, and is set [here](https://github.com/eml-eda/plinio/blob/2e1a4353d7ce16106e91f9a6ac5991a1a7eed46a/plinio/methods/mixprec/mixprec.py#L257).

The sampling strategy for the quantization parameters is instead set with [this function](https://github.com/eml-eda/plinio/blob/2e1a4353d7ce16106e91f9a6ac5991a1a7eed46a/plinio/methods/mixprec/nn/mixprec_qtz.py#L142).
